src/panels/lovelace/common/handle-click.ts

```diff
@@ -3,20 +3,30 @@ import { LovelaceElementConfig } from "../elements/types";
 import { fireEvent } from "../../../common/dom/fire_event";
 import { navigate } from "../../../common/navigate";
 import { toggleEntity } from "../../../../src/panels/lovelace/common/entity/toggle-entity";
-import { LovelaceCardConfig } from "../../../data/lovelace";
+import { LovelaceCardConfig, ActionConfig } from "../../../data/lovelace";
+import { EntityConfig } from "../entity-rows/types";
+
+export interface ConfigEntity extends EntityConfig {
+  tap_action?: ActionConfig;
+  hold_action?: ActionConfig;
+}
 
 export const handleClick = (
   node: HTMLElement,
   hass: HomeAssistant,
-  config: LovelaceElementConfig | LovelaceCardConfig,
+  config: LovelaceElementConfig | LovelaceCardConfig | ConfigEntity,
   hold: boolean
 ): void => {
-  let action = config.tap_action || "more-info";
-
+  let actionConfig;
   if (hold && config.hold_action) {
-    action = config.hold_action;
+    actionConfig = config.hold_action;
+  } else if (!hold && config.tap_action) {
+    actionConfig = config.tap_action;
   }
 
+  const action =
+    actionConfig && actionConfig.action ? actionConfig.action : "more-info";
```

Comment:
Not all the types handled by `config` have an `action` field. I don't think any elements are likely working right now with this

Reply:
Thats why it has the check for action field

Reply:
But you can still define an action type for the elements beyond "more-info". That is lost here.

Reply:
Perhaps we should just align the elements now as well? `icon`, `image`, `state-icon` and `state-label`

Reply:
Oh you are saying the elements of picture elements
Yea I havent gotten there yet. Next commit. Looking at that now